src/k8s/pkg/k8sd/app/hooks_start.go

```diff
@@ -13,7 +14,20 @@ func onStart(s *state.State) error {
 	snap := snap.SnapFromContext(s.Context)
 
 	configController := controllers.NewNodeConfigurationController(snap, func(ctx context.Context) *k8s.Client {
-		return k8s.RetryNewClient(ctx, snap.KubernetesNodeRESTClientGetter("kube-system"))
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-time.After(3 * time.Second):
+			default:
```

Comment:
related, since we are doing fixes on this PR. the default case here will ignore the `time.After` above. remove it